cmds/core/shasum/shasum.go

```diff
@@ -60,7 +71,7 @@ func shasum(w io.Writer, r io.Reader, args ...string) error {
 			return err
 		}
 		defer file.Close()
-		if hashbytes, err = shaGenerator(w, file, algorithm); err != nil {
+		if hashbytes, err = shaGenerator(file, Algorithm(algorithm)); err != nil {
```

Comment:
but it does depend on people knowing the relation of the constant to the const value, which seems not as simple. But I guess OK.